hibernate-core/src/main/java/org/hibernate/dialect/CockroachDialect.java

```diff
@@ -148,10 +157,10 @@ public CockroachDialect(DatabaseVersion version, PostgreSQLDriverKind driverKind
 		this.driverKind = driverKind;
 	}
 
-	protected static DatabaseVersion fetchDataBaseVersion( DialectResolutionInfo info ) {
+	protected static DatabaseVersion fetchDataBaseVersion(DialectResolutionInfo info) {
```

Comment:
I wonder if we should handle this in `org.hibernate.dialect.Database#createDialect` for CockroachDB instead.  At the end of the day, its just about resolving the version to pass into the constructor.  

Also, it seens like this misses the case of an explicitly configured version?  I.e. `jakarta.persistence.database-product-version`, `jakarta.persistence.database-major-version`, `jakarta.persistence.database-minor-version`

Reply:
I noticed this too, we're actually querying CockroachDB twice on startup to fetch the `version()` String, but I wasn't sure if a change in the constructors would be correct. I've added a `CockroachDialect(DialectResolutionInfo info, String versionString)` to avoid this.

Regarding the explicitly configured versions, I've also added `info.makeCopy()` as an additional fallback if no version can be retrieved from the database or specified through the new custom parameter (though in this case we would just be resolving `PostgreSQLDialect` I suppose).